app/v2/notifications/post_notifications.py

```diff
@@ -312,84 +312,6 @@ def process_document_uploads(
     return personalisation_data
 
 
-def process_letter_notification(
-    *,
-    letter_data,
-    api_key,
-    template,
-    reply_to_text,
-    precompiled=False,
-):
-    if api_key.key_type == KEY_TYPE_TEAM:
-        raise BadRequestError(message='Cannot send letters with a team api key', status_code=403)
-
-    if not api_key.service.research_mode and api_key.service.restricted and api_key.key_type != KEY_TYPE_TEST:
-        raise BadRequestError(message='Cannot send letters when service is in trial mode', status_code=403)
-
-    if precompiled:
-        return process_precompiled_letter_notifications(
-            letter_data=letter_data, api_key=api_key, template=template, reply_to_text=reply_to_text
-        )
-
-    test_key = api_key.key_type == KEY_TYPE_TEST
-
-    # if we don't want to actually send the letter, then start it off in SENDING so we don't pick it up
-    status = NOTIFICATION_CREATED if not test_key else NOTIFICATION_SENDING
-    queue = QueueNames.CREATE_LETTERS_PDF if not test_key else QueueNames.RESEARCH_MODE
-
-    notification = create_letter_notification(
-        letter_data=letter_data, template=template, api_key=api_key, status=status, reply_to_text=reply_to_text
-    )
-
-    create_letters_pdf.apply_async([str(notification.id)], queue=queue)
-
-    if test_key:
-        if current_app.config['NOTIFY_ENVIRONMENT'] in ['preview', 'development']:
-            create_fake_letter_response_file.apply_async((notification.reference,), queue=queue)
-        else:
-            update_notification_status_by_reference(notification.reference, NOTIFICATION_DELIVERED)
-
-    return notification
-
-
-def process_precompiled_letter_notifications(
-    *,
-    letter_data,
-    api_key,
-    template,
-    reply_to_text,
-):
-    try:
-        status = NOTIFICATION_PENDING_VIRUS_CHECK
-        letter_content = base64.b64decode(letter_data['content'])
-    except ValueError:
-        raise BadRequestError(message='Cannot decode letter content (invalid base64 encoding)', status_code=400)
-
-    notification = create_letter_notification(
-        letter_data=letter_data, template=template, api_key=api_key, status=status, reply_to_text=reply_to_text
-    )
-
-    filename = upload_letter_pdf(notification, letter_content, precompiled=True)
```

Comment:
Can `from app.letters.utils import upload_letter_pdf.py` `upload_letter_pdf` be deleted now that we are not using it in this file? 

It looks like `upload_letter_pd` is only used here and in tests.

Reply:
I've removed this code.